doc/vfd-swmr-user-guide.md

```diff
@@ -0,0 +1,496 @@
+between writer and readers.  On a POSIX system, the metadata file
+may be placed on any *local* filesystem that the reader and writer
+share.  The `md_file_path` parameter tells where to put the metadata
+file.
+
+The `md_pages_reserved` parameter tells how many pages to reserve
+at the beginning of the metadata file for the metadata-file header
+and the metadata index.  The header has an entire page to itself.
+The remaining `md_pages_reserved - 1` pages are reserved for the
+metadata index.  If the index grows larger than its initial
+allocation, then it will move to a new location in the metadata file,
+and the initial allocation will be reclaimed.  `md_pages_reserved`
+must be at least 2.
+
+The `version` parameter tells what version of VFD SWMR configuration
+the parameter struct `config` contains.  For now, it should be
+initialized to `H5F__CURR_VFD_SWMR_CONFIG_VERSION`.
+
+## Using virtual datasets (VDS)
+
+An application may want to use VFD SWMR to create, read, or write
+a virtual dataset.  Unfortunately, VDS does not work properly with
+VFD SWMR at this time.  In this section, we describe some workarounds
+that can be used with great care to make VDS and VFD SWMR cooperate.
+
+A virtual dataset, when it is read or written, will open files on
+an application's behalf in order to access the source datasets
+inside.  If a virtual dataset resides on file `v.h5`, and one of
+its source datasets resides on a second file, `s1.h5`, then the
+virtual dataset will try to open `s1.h5` using the same file-access
+properties as `v.h5`.  Thus, if `v.h5` is open with VFD SWMR with
+metadata file `v.shadow`, then the virtual dataset will try to open
+`s1.h5` with the same metadata file, which will fail.
+
+Suppose that `v.h5` is *not* open with VFD SWMR, but it was opened
+with default file-access properties.  Then the virtual dataset will
+open the source dataset on `s1.h5` with default file-access
+properties, too.  This default virtual-dataset behavior is not
+helpful to the application that wants to use VFD SWMR to read or
+write source datasets.
+
+To use VFD SWMR with VDS, an application should *pre-open* each file
+using its preferred file-access properties, including independent metadata
+filenames for each source file.  As long as the virtual dataset remains
+in use, the application should leave each of the pre-opened files open.
+In this way the library, when it tries to open the source files, will
+always find them already open and re-use the already-open files with the
+file-access properties established on first open.
+
+## Pushing HDF5 raw data to reader visibility
+
+At present, VFD SWMR is hard coded to flush raw data at the end of 
+each tick.  While this imposes additional overhead, it simplifies testing, 
+and is probably desirable for applications that do not require the best
+possible raw data throughput.  We plan to upgrade our tests and make this 
+user configurable in the first production release.
```

Comment:
Did this get finished?

Reply:
Yes it is user configurable by setting the "flush_raw_data" field in H5F_vfd_swmr_config_t via H5Pset_vfd_swmr_config() API.